Add COSMOS Event Routing

Jbsco opened the pull request:
This pull request should follow #18. These changes provide the inter-connectivity required to parse event packets from the Adamant assembly using the `socket_event_decoder.py` script included with Adamant.

To do this a router is added to the Adamant assembly's COSMOS plugin, which allows parsing of event packets from within the Adamant environment on port 7779. This was tested and found functional on Linux.

Jbsco committed:
Update Docker configuration and readme to support packet event decoding

Jbsco committed:
Implement COSMOS router in plugin configurations for Linux and Pico examples

Jbsco commented:
I've re-based this branch on the upstream after merging #18.

dinkelk commented:
Running `create_container.sh` with this PR breaks `python` for me:

```
(.py_env) user@30bb68c8e481:~$ python
Fatal Python error: init_sys_streams: can't initialize sys standard streams
Python runtime state: core initialized
Traceback (most recent call last):
  File "/usr/lib/python3.10/io.py", line 52, in <module>
  File "/home/user/adamant_example/gnd/cosmos/venv/lib/python3.11/site-packages/pip/_vendor/rich/abc.py", line 1, in <module>
ImportError: cannot import name 'ABC' from partially initialized module 'abc' (most likely due to a circular import) (/home/user/adamant_example/gnd/cosmos/venv/lib/python3.11/site-packages/pip/_vendor/rich/abc.py)
```

I doubt this is the PR's fault though. Will need to dig into this deeper. It is time to update to the new Ubuntu version anyways, so maybe that will fix it.

dinkelk committed:
Add comments for COSMOS plugin.txt to show purpose of router

dinkelk committed:
Add more context to Pico README.md about event viewer

dinkelk committed:
Add COSMOS packet router to the Mac-specific plugin

Jbsco committed:
Update Pico readme to include Windows in serial bridge instructions

Jbsco commented:
Tested and found functional on Windows using the `serial_tcp_bridge.py` script.

dinkelk committed:
Apply fixes to Pico README.md

Fixes to Pico README.md

Apply additional fixes to Pico README.md

Apply fixes for Linux and Pico README.md
